File: python/simplexy.py
#
# Very simple CSV plotting example for Python.
#

# Import the modules
import array
import numpy as np
import matplotlib.pyplot as plt

F      = '../data/1430258910-example.csv'
result = np.genfromtxt(F, delimiter=',')

# Isolate variables you want to use
unixtime  = result[:,1] # Unix time stamp
dat_tempc = result[:,9] # Temperature (C)

# Create time offsets
nprofiles = int(max(unixtime) - min(unixtime))
time_offset = np.arange(nprofiles)
time_scaled = np.arange(len(dat_tempc))


# Draw plot
plt.figure()
plt.plot(time_scaled, dat_tempc,'r-')

x1,x2,y1,y2 = plt.axis()
plt.axis((x1,x2,10,35))
plt.show()

File: matlab/simplexy.m
close all;
clear all;

f = '../data/1430258910-example.csv';

header_lines = 1;

unix_time = get_csvcol(f, header_lines, 1);
dat_tempc = get_csvcol(f, header_lines, 9);

dummyx = (unix_time/86400 + datenum(1970,1,1));


% Create the plot
hold on;
box on;
plot(dummyx, dat_tempc,'-k')
set(gca, 'YLim', [10, 35])
title('Time vs. Temperature ', 'FontSize', 16)
xlabel('Time (s)', 'FontSize', 14)
ylabel('Temperature (C) ', 'FontSize', 14)

datetick('x','MM:SS')

%set(gca, 'XTick', dummyx);
%set(gca, 'XTickLabel', labels);

hold off;